tests/Tracing/DynamicSamplingContextTest.php

```diff
@@ -0,0 +1,176 @@
+        ?string $expectedTransaction
+    ): void {
+        $dsc = DynamicSamplingContext::fromHeader($header);
+
+        $this->assertSame($expectedTraceId, $dsc->get('trace_id'));
+        $this->assertSame($expectedPublicKey, $dsc->get('public_key'));
+        $this->assertSame($expectedSampleRate, $dsc->get('sample_rate'));
+        $this->assertSame($expectedRelease, $dsc->get('release'));
+        $this->assertSame($expectedEnvironment, $dsc->get('environment'));
+        $this->assertSame($expectedUserSegment, $dsc->get('user_segment'));
+        $this->assertSame($expectedTransaction, $dsc->get('transaction'));
+    }
+
+    public function fromHeaderDataProvider()
+    {
+        yield [
+            '',
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+        ];
+
+        yield [
+            'sentry-trace_id=d49d9bf66f13450b81f65bc51cf49c03,sentry-public_key=public,sentry-sample_rate=1',
+            'd49d9bf66f13450b81f65bc51cf49c03',
+            'public',
+            '1',
+            null,
+            null,
+            null,
+            null,
+        ];
+
+        yield [
+            'sentry-trace_id=d49d9bf66f13450b81f65bc51cf49c03,sentry-public_key=public,sentry-sample_rate=1,sentry-release=1.0.0,sentry-environment=test,sentry-user_segment=my_segment,sentry-transaction=<unlabeled transaction>',
+            'd49d9bf66f13450b81f65bc51cf49c03',
+            'public',
+            '1',
+            '1.0.0',
+            'test',
+            'my_segment',
+            '<unlabeled transaction>',
+        ];
+    }
+
+    public function testFromTransaction(): void
+    {
+        $client = $this->createMock(ClientInterface::class);
+        $client->expects($this->once())
+            ->method('getOptions')
+            ->willReturn(new Options([
+                'dsn' => 'http://public@example.com/sentry/1',
+                'release' => '1.0.0',
+                'environment' => 'test',
+            ]));
+
+        $hub = $this->createMock(HubInterface::class);
+        $hub->expects($this->once())
+            ->method('getClient')
+            ->willReturn($client);
+
+        $transactionContext = new TransactionContext();
+        $transactionContext->setName('foo');
+
+        $transaction = new Transaction($transactionContext, $hub);
+
+        $dsc = DynamicSamplingContext::fromTransaction($transaction, $hub);
+
+        $this->assertSame((string) $transaction->getTraceId(), $dsc->get('trace_id'));
+        $this->assertSame((string) $transaction->getMetaData()->getSamplingRate(), $dsc->get('sample_rate'));
+        $this->assertSame('foo', $dsc->get('transaction'));
+        $this->assertSame('public', $dsc->get('public_key'));
+        $this->assertSame('1.0.0', $dsc->get('release'));
+        $this->assertSame('test', $dsc->get('environment'));
+        $this->assertSame('my_segment', $dsc->get('user_segment'));
```

Comment:
I'm not sure how to properly mock the scope in this test. Any help is kindly appreciated.